microsoft/testsuites/nested/nested.py

```diff
@@ -45,34 +59,37 @@ def verify_nested_kvm_basic(
             nested_image_url,
         ) = parse_nested_image_variables(variables)
 
-        # get l2 vm
-        l2_vm = qemu_connect_nested_vm(
-            node,
-            nested_image_username,
-            nested_image_password,
-            nested_image_port,
-            nested_image_url,
-        )
+        try:
+            # get l2 vm
+            l2_vm = qemu_connect_nested_vm(
+                node,
+                nested_image_username,
+                nested_image_password,
+                nested_image_port,
+                nested_image_url,
+            )
 
-        # verify file is correctly copied from L1 VM to L2 VM
-        node.tools[Echo].write_to_file(
-            NESTED_VM_TEST_FILE_CONTENT,
-            node.get_pure_path(NESTED_VM_TEST_FILE_NAME),
-        )
-        node.tools[Sshpass].copy(
-            NESTED_VM_TEST_FILE_NAME,
-            NESTED_VM_TEST_FILE_NAME,
-            node.public_address,
-            nested_image_username,
-            nested_image_password,
-            nested_image_port,
-        )
+            # verify file is correctly copied from L1 VM to L2 VM
+            node.tools[Echo].write_to_file(
+                NESTED_VM_TEST_FILE_CONTENT,
+                node.get_pure_path(NESTED_VM_TEST_FILE_NAME),
+            )
+            node.tools[Sshpass].copy(
+                NESTED_VM_TEST_FILE_NAME,
+                NESTED_VM_TEST_FILE_NAME,
+                node.public_address,
+                nested_image_username,
+                nested_image_password,
+                nested_image_port,
+            )
 
-        uploaded_message = l2_vm.tools[Cat].read(NESTED_VM_TEST_FILE_NAME)
-        assert_that(
-            uploaded_message,
-            "Content of the file uploaded to L2 vm from L1 should match",
-        ).is_equal_to(NESTED_VM_TEST_FILE_CONTENT)
+            uploaded_message = l2_vm.tools[Cat].read(NESTED_VM_TEST_FILE_NAME)
+            assert_that(
+                uploaded_message,
+                "Content of the file uploaded to L2 vm from L1 should match",
+            ).is_equal_to(NESTED_VM_TEST_FILE_CONTENT)
 
-        # verify that files could be downloaded from internet on L2 VM
-        l2_vm.tools[Wget].get(NESTED_VM_TEST_PUBLIC_FILE_URL)
+            # verify that files could be downloaded from internet on L2 VM
+            l2_vm.tools[Wget].get(NESTED_VM_TEST_PUBLIC_FILE_URL)
+        finally:
+            node.tools[Qemu].stop_vm()
```

Comment:
Should it delete the vm as well? This logic can be put into after_case.

Reply:
This kills the l2 vm which was created.

Reply:
If so, how about rename the method to `delete_vms`?